internal/services/compute/shared_image_resource.go

```diff
@@ -438,3 +429,48 @@ func flattenGalleryImagePurchasePlan(input *compute.ImagePurchasePlan) []interfa
 		},
 	}
 }
+
+func expandGalleryImageFeatures(d *pluginsdk.ResourceData) *[]compute.GalleryImageFeature {
+	var features []compute.GalleryImageFeature
+
+	if d.Get("trusted_launch_enabled").(bool) {
+		features = append(features, compute.GalleryImageFeature{
+			Name:  utils.String("SecurityType"),
+			Value: utils.String("TrustedLaunch"),
+		})
+	}
+
+	if d.Get("support_accelerated_network").(bool) {
+		features = append(features, compute.GalleryImageFeature{
+			Name:  utils.String("IsAcceleratedNetworkSupported"),
+			Value: utils.String("true"),
+		})
+	}
+
+	return &features
+}
+
+func flattenAndSetGalleryImageFeatures(d *pluginsdk.ResourceData, features *[]compute.GalleryImageFeature) {
+	trustedLaunchEnabled := false
+	supportAcceleratedNetwork := false
+
+	if features != nil {
+		featuresMap := make(map[string]string, len(*features))
+		for _, feature := range *features {
+			if feature.Name != nil && feature.Value != nil {
+				featuresMap[*feature.Name] = *feature.Value
+			}
+		}
+
+		if featuresMap["SecurityType"] == "TrustedLaunch" {
+			trustedLaunchEnabled = true
+		}
+
+		if strings.EqualFold(featuresMap["IsAcceleratedNetworkSupported"], "true") {
+			supportAcceleratedNetwork = true
+		}
```

Comment:
```suggestion
		for _, feature := range *features {
			if feature.Name == nil || feature.Value == nil {
				continue
			}
			
			if strings.EqualFold(*feature.Name, "SecurityType") {
				trustedLaunchEnabled = strings.EqualFold(*feature.Value, "TrustedLaunch")
			}
			
			if strings.EqualFold(*feature.Name, "IsAcceleratedNetworkSupported") {
				supportAcceleratedNetwork = strings.EqualFold(*feature.Value, "true")
			}
		}
```

Reply:
Updated. Looks more straightforward since we will probably not having too many features here